src/components/store/types/Reducer.ts

```diff
@@ -0,0 +1,7 @@
+import { EditorState } from './EditorState';
```

Comment:
we store all types at the `types` directory. Maybe it would better to place the Store types same way?

```
/types/
  /store/
    ...
```